yearn/deploy/01_deploy_convex_strategy.ts

```diff
@@ -0,0 +1,42 @@
+import { HardhatRuntimeEnvironment } from "hardhat/types"
+import { DeployFunction } from "hardhat-deploy/types"
+
+const func: DeployFunction = async function (hre: HardhatRuntimeEnvironment) {
+  const { deployments, getNamedAccounts, helpers } = hre
+  const { deploy, log } = deployments
+  const { deployer } = await getNamedAccounts()
+
+  const TBTCCurveVault = await deployments.getOrNull("TBTCCurveVault")
+  if (!TBTCCurveVault || !helpers.address.isValid(TBTCCurveVault.address)) {
+    throw new Error("Address of TBTCCurveVault must be set")
+  }
+
+  const TBTCCurvePoolDepositor = await deployments.getOrNull("TBTCCurvePoolDepositor")
+  if (!TBTCCurvePoolDepositor || !helpers.address.isValid(TBTCCurvePoolDepositor.address)) {
+    throw new Error("Address of TBTCCurvePoolDepositor must be set")
+  }
+
+  const TBTCConvexRewardPool = await deployments.getOrNull("TBTCConvexRewardPool")
+  if (!TBTCConvexRewardPool || !TBTCConvexRewardPool.linkedData.id) {
+    throw new Error("ID of TBTCConvexRewardPool must be set")
+  }
+
+  log(`tbtcCurveVault: ${TBTCCurveVault.address}`)
+  log(`tbtcCurvePoolDepositor: ${TBTCCurvePoolDepositor.address}`)
+  log(`tbtcConvexRewardPoolId: ${TBTCConvexRewardPool.linkedData.id}`)
```

Comment:
What is the story behind the id? Is it a property that can be pulled from a contract with a get function or some fixed value? 
I just want to make sure it doesn't get lost in the external artifact that we want to store and we're sure it will be there.

Reply:
This is a fixed value. Long story short, there is a Convex booster contract which is a kind of reward pool manager. Each registered reward pool gets its own ID used for further identification. It is simpler to operate knowing the pool ID than its address so this is why it is here. I know passing it using `linkedData` looks a bit strange at first sight. However, I think it's better than using an env variable and it is consistent with the way we deal with external dependencies.